documentation/projects/proposals/data_normalization/20240227-implementation_plan_catalog_data_cleaning.md

```diff
@@ -0,0 +1,198 @@
+# 2024-02-27 Implementation Plan: Catalog Data Cleaning
+
+**Author**: @krysal
+
+## Reviewers
+
+- [ ] @obulat
+- [ ] @AetherUnbound
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/430)
+
+This project does not have a project proposal because the scope and rationale of
+the project are clear, as defined in the project thread. In doubt, check the
+[Expected Outcomes](#expected-outcomes) section below.
+
+## Overview
+
+This document describes a solution for incorrect data in the catalog database
+(DB) that has to be cleaned up every time a data refresh is run, avoiding wasted
+resources.
+
+## Background
+
+One of the steps of the [data refresh process for images][img-data-refresh] is
+cleaning the data that is not fit for production. This process is triggered
+weekly by an Airflow DAG, and then runs in the Ingestion Server, taking
+approximately just over **20 hours** to complete, according to a inspection of
+latest executions. The cleaned data is only saved to the API database, which is
+replaced each time during the same data refresh, causing it to have to be
+repeated each time to make the _same_ corrections.
+
+This cleaning process was designed this way to speed the rows update up since
+the relevant part was to provide the correct data to users via the API. Most of
```

Comment:
Phrasing:
```suggestion
This cleaning process was designed this way to optimize writes to the catalog, since
the most important factor was to provide the correct data to users via the API. Most of
```

Reply:
I updated the suggestion to use "upstream database" instead, does it look okay now?

Reply:
I agree that this sentence needs clarification since it's not clear if we mean "catalog db rows update" or the "API db rows update".

Reply:
I see! In this case, it should be the API DB then, because the Ingestion Serves saves the cleaned data there.

Applied 👍